diff --git a/config.def.h b/config.def.h
index 6b68b5c..3c7d9d4 100644
--- a/config.def.h
+++ b/config.def.h
@@ -23,3 +23,6 @@ static unsigned int lines      = 0;
  * for example: " /?\"&[]"
  */
 static const char worddelimiters[] = " ";
+
+/* Size of the window border */
+static unsigned int border_width = 0;
diff --git a/dmenu.c b/dmenu.c
index 29dd829..ee86d42 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -698,6 +698,10 @@ setup(void)
 	win = XCreateWindow(dpy, parentwin, x, y, mw, mh, 0,
 	                    CopyFromParent, CopyFromParent, CopyFromParent,
 	                    CWOverrideRedirect | CWBackPixel | CWEventMask, &swa);
+	if (border_width){
+        XSetWindowBorderWidth(dpy, win, border_width);
+        XSetWindowBorder(dpy, win, scheme[SchemeSel][ColBg].pixel);
+    }
 	XSetClassHint(dpy, win, &ch);
 
 
@@ -770,6 +774,11 @@ main(int argc, char *argv[])
 			colors[SchemeSel][ColFg] = argv[++i];
 		else if (!strcmp(argv[i], "-w"))   /* embedding window id */
 			embed = argv[++i];
+		else if (!strcmp(argv[i], "-bw")){
+            int tmp = atoi(argv[++i]);
+            if (tmp > 0)
+                border_width = tmp;
+        }
 		else
 			usage();
 
