diff --git a/dmenu.c b/dmenu.c
index 2de5e5e..0489628 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -357,15 +357,20 @@ movewordedge(int dir)
 static void
 keypress(XKeyEvent *ev)
 {
-	char buf[64];
+	char *buf;
+    size_t buf_size = 64;
+    buf = (char *)malloc(buf_size);
 	int len;
 	KeySym ksym = NoSymbol;
 	Status status;
 
-	len = XmbLookupString(xic, ev, buf, sizeof buf, &ksym, &status);
+	len = XmbLookupString(xic, ev, buf, buf_size - 1, &ksym, &status);
 	switch (status) {
 	default: /* XLookupNone, XBufferOverflow */
 		return;
+    case XBufferOverflow:
+        buf = realloc(buf, len + 1);
+        len = XmbLookupString(xic, ev, buf, len, &ksym, &status);
 	case XLookupChars: /* composed string from input method */
 		goto insert;
 	case XLookupKeySym:
@@ -558,6 +563,7 @@ insert:
 	}
 
 draw:
+    free(buf);
 	drawmenu();
 }
 
