diff --git a/src/video/VideoFrame.cpp b/src/video/VideoFrame.cpp
index f557f3a..659a52e 100644
--- a/src/video/VideoFrame.cpp
+++ b/src/video/VideoFrame.cpp
@@ -46,10 +46,6 @@ VideoFrame::VideoFrame( VideoFrame&& other ) : frame( other.frame ), context( ot
 bool VideoFrame::is_keyframe() const{ return frame->key_frame; }
 
 int VideoFrame::frameNumber() const{
-	//TODO: Check if this ever contains useful info
-	if (frame->display_picture_number != 0)
-		return frame->display_picture_number;
-	
 	//Calculate it based on the frame time
 	auto second = frame->pts / (double)AV_TIME_BASE * 1000;
 	auto frame = second * av_q2d(context.framerate);
